Controllers/RoomsController.cs

```diff
@@ -97,21 +78,15 @@ public async Task<ActionResult<Room>> PostRoom(Room room)
         [HttpDelete("{id}")]
         public async Task<IActionResult> DeleteRoom(int id)
         {
-            var room = await _context.Rooms.FindAsync(id);
-            if (room == null)
-            {
-                return NotFound();
-            }
-
-            _context.Rooms.Remove(room);
-            await _context.SaveChangesAsync();
-
-            return NoContent();
+            await _roomRepository.DeleteRoom(id);
+            
+                return NotFound();  
         }
 
-        private bool RoomExists(int id)
+        private async Task<bool> RoomExists(int id)
         {
-            return _context.Rooms.Any(e => e.Id == id);
+            return await _roomRepository.RoomExists(id);
         }
```

Comment:
I would not expect you need to keep this. All database context stuff should live in the `RoomRepository`.